From 181db7f7e861deec2bcf7babe3c96e064acfea0b Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:52 -0400
Subject: [ethernet] bnx2x: fix set_setting for some PHYs

Message-id: <1409946615-5862-48-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91588
O-Subject: [RHEL7.1 PATCH 47/70] bnx2x: fix set_setting for some PHYs
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 042d7654ecb68383c392e3c7e86e04de023f65f9
Author: Yaniv Rosner <Yaniv.Rosner@qlogic.com>
Date:   Wed Jul 23 22:12:57 2014 +0300

    bnx2x: fix set_setting for some PHYs

    Allow set_settings() to complete succesfully even if link is
    not estabilished and port type isn't known yet.

    Signed-off-by: Yaniv Rosner <Yaniv.Rosner@qlogic.com>
    Signed-off-by: Dmitry Kravkov <Dmitry.Kravkov@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
index 9d9a7ec..a1d67f1 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
@@ -416,6 +416,7 @@ static int bnx2x_set_settings(struct net_device *dev, struct ethtool_cmd *cmd)
    break;
   case PORT_FIBRE:
   case PORT_DA:
+  case PORT_NONE:
    if (!(bp->port.supported[0] & SUPPORTED_FIBRE ||
          bp->port.supported[1] & SUPPORTED_FIBRE)) {
     DP(BNX2X_MSG_ETHTOOL,
-- 
1.7.1